### PR TITLE
srvc: 0.8.0 -> 0.9.0

### PR DESCRIPTION
**Pull request by john-shaffer**
###### Description of changes

- Add changelog
- Add webfs check input for use by the tests
- Pass env var to disable test timeout. Remove the patch that did this.
- https://github.com/insilica/rs-srvc/blob/v0.9.0/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/srvc/default.nix [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-    Security
-  ];
+  # remove timeouts in tests to make them less flaky
+  TEST_SRVC_DISABLE_TIMEOUT = 1;
 
   meta = with lib; {
+    changelog =
+      "https://github.com/insilica/rs-srvc/blob/v${version}/CHANGELOG.md";
     description = "Sysrev version control";
     homepage = "https://github.com/insilica/rs-srvc";
```
Comment (figsoda) [inline pkgs/applications/version-management/srvc/default.nix:26]:
```suggestion
    description = "Sysrev version control";
    homepage = "https://github.com/insilica/rs-srvc";
    changelog = "https://github.com/insilica/rs-srvc/blob/v${version}/CHANGELOG.md";
```

**Review by figsoda — COMMENTED** [review]
just some nitpicks

File: pkgs/applications/version-management/srvc/default.nix [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
+    sha256 = "sha256-XslMwA1DhztK9DPNCucUpzjCQXz6PN8ml8JBvKtJeqg=";
   };
 
-  cargoSha256 = "sha256-nJM7/w4awbCZQysUOSTO6bfsBXO3agJRdp1RyRZhtUY=";
+  cargoSha256 = "sha256-KxwBF5t8lcaH8ZD4SorIBiq7p6r9LGHfEOyNXtB9HJw=";
 
-  # remove timeouts in tests to make them less flaky
-  patches = [ ./tests-no-timeout.patch ];
+  buildInputs = lib.optionals stdenv.isDarwin [ Security ];
+  nativeBuildInputs = [ webfs ];
```
[the remark this answers is not in the document]
Comment (john-shaffer) [reply pkgs/applications/version-management/srvc/default.nix]: Right, thanks for pointing that out

**Comment by john-shaffer** [comment]
Thanks. I've updated with those changes

**Review by figsoda — APPROVED** [review]

**Pull request merged**